### ISSUE TITLE
issues with https?

### ISSUE THREAD

**Issue by morrow95** [getgrav/grav-plugin-simplesearch#88]
If I force https (which is what the site should be) then searching no longer works.  I am getting this error on searches in FireFox : 

```
The address wasn’t understood

Firefox doesn’t know how to open this address, because one of the following protocols (undefined) isn’t associated with any program or is not allowed in this context.

    You might need to install other software to open this address.
```
The url in the browser when this happens shows ```undefined:undefined``` as the address.

What is the deal here and how can I fix this?  I am assuming somewhere it is trying to use http or something, but I could be wrong.

**Comment by morrow95** [comment]
More testing and I have no clue what is causing this.  I did find this interesting though : 

I can manually enter in a search url in my browser and it will work correctly.  Using the text input on the page gives me the error I mentioned in the op... manually entering a search url in the browser like https://blog.example.com/search:donec will work correctly.

What the heck is the problem here?

**Comment by flaviocopes** [comment]
Looks like a server configuration issue because I can't replicate the problem. Is your web server apache, and are you using the .htaccess file provided?

Are you entering the search term in the https page, or in an http page that's then forced to https?

Are you forcing https by using the System config setting? You might also want to try this at the .htaccess level instead of the config option, which prevents useless processing at the application level (Grav)

**Comment by morrow95** [comment]
"Is your web server apache, and are you using the .htaccess file provided?"

yes and yes

"Are you entering the search term in the https page, or in an http page that's then forced to https?"

That wouldn't be possible because everything is forced to https so you will always be doing so from an https page, but the short answer is yes.  

"Are you forcing https by using the System config setting?"

I am, however, even if I turn this option off the page still does not work.  The only time it works is with a clean install and by clean I mean the grav-skeleton-blog-site-v1.1.1 with the admin plugin installed through the cl.  I am also using my own theme which is setup as a theme inheritance of the antimatter theme, BUT even if I switch back to antimatter it doesn't work.

First thing is I don't understand why it works then just stops all of a sudden after I make a change in the site config.... and why does it not work again when I change the site config back to defaults?  Second, why does it work when I manually enter the url in the browser?  It seems as though the js of the plugin is the problem here since entering the url directly into the browser works.

I will try and do https through htaccess myself as a test later on.  I am also considering the possibility that this is some weird caching issue.  I have turned caching off, and tried all sorts of combinations of settings (even the defaults which work fine to start) and nothing does the trick.

**Comment by morrow95** [comment]
@flaviocopes 

I can replicate this issue whenever I want right now.

In the admin plugin I am going to simplesearch's plugin settings.  I am enabling 'Display Search Button' which does exactly like it should, BUT if I disable 'Use built in CSS' I get the protocol error.  I can then enable it, reload the page, and search works again.  I can repeat this over and over.

The only other settings I have changed are for /user/config/system.yaml which would be : 

Hide home route in URLs set to true
expires set to 5 (just to ensure changes are happening on the pages during my tests)
everything else is the default for the grav-skeleton-blog-site-v1.1.1 build with the admin plugin installed.

With that said, what is happening here when I disable using the plugin's css?  The only thing that should do is prevent ```<link href="/user/plugins/simplesearch/css/simplesearch.css" type="text/css" rel="stylesheet" />``` from being loaded into the head of the theme, but there must be something else happening here.  I have no need for that file because I am using my own css styling for it.

**Comment by morrow95** [comment]
Okay, it isn't the css it is the button.  Enter key works fine and the button click does not.  I had to modify the on click handler for the button and now it is working properly.  The issue was due to the fact I slightly modified the ```simplesearch_searchbox.html.twig```.  Should've seen this one earlier, but forgot all about the change to that twig.

Sorry for the hassle and thanks for looking.  Have a great week!
